modules/edit-modes/src/lib/geojson-edit-mode.js

```diff
@@ -300,67 +271,78 @@ export class BaseGeoJsonEditMode extends BaseEditMode<FeatureCollection, ModeHan
     return this.getAddFeatureAction(geometry);
   }
 
-  handleClick(event: ClickEvent): void {
-    const editAction = this.handleClickAdapter(event);
+  // TODO: factor out all the duplicate calls to setFeatureCollection
+
+  handleClick(event: ClickEvent, props: ModeProps<FeatureCollection>): void {
+    this.setFeatureCollection(props.data);
```

Comment:
Why you need a local copy of `props.data` as `featureCollection`?

Reply:
`this.featureCollection` is an instance of `ImmutableFeatureCollection`. In a future diff, I'll make it more stateless. But there is lots of code in the `ModeHandler`s that does `this.getFeatureCollection()`. So, just trying to make the refactor a little easier.